test/libsolidity/semanticTests/literal_suffixes/intermediate.sol

```diff
@@ -0,0 +1,21 @@
+type Length is uint;
+
+function km(uint x) pure returns (Length) { return Length.wrap(x * 1000); }
+
+struct Float {
+    uint mantissa;
+    int exponent;
+}
+
+function f(uint mat, int exp) pure returns (Float memory) {
+  return Float(mat, exp);
+}
+contract C {
+  Length public length = 1000 km;
+  Float public factor = 1.23 f;
```

Comment:
So is the dots delimiting the arguments in terms of user defined types?

Reply:
Sorry, I don't understand your question. There is only one dot, and `1.23 f` is only related to user defined value types in that `f` returns one, but you don't need this concept for the example.

Reply:
I believe the question is whether `1` and `23` will be the arguments for `f(uint mat, int exp)`. 
If so, the example is maybe a bit confusing because `1.23` looks very much like the decimal `1.23` and not like mantissa/exponent...

Reply:
`1.23` is the same as `123e-2`. This line will result in `f(123, -2)` being called.

Reply:
Aah, is that simply the mechanism of how the suffixes work with decimals then?

Reply:
Yes.